Update aws-add-remove-elbv2.json

phillip-haydon opened the pull request:
When on Linux arm64 and installing the aws module it uses AWSPowerShell.NetCore, but on Windows with the older version it could be AWSPowerShell. So this checks to see if either version was automatically imported.

If it's already imported then it carries on, otherwise it attempts to see if it's installed and then import it. Or throw a message to say either version cannot be found.

```
if (Get-Module | Where-Object { $_.Name -like "AWSPowerShel*" }) {
	Write-Host "AWS PowerShell module is already loaded."
} else {
    if (!(Get-Module -ListAvailable | Where-Object {  $_.Name -like "AWSPowerShel*" })) {
    	Write-Host "AWSPowerShell / AWSPowerShell.NetCore not found"
    } else {
    	Get-Module -ListAvailable | Where-Object {  $_.Name -like "AWSPowerShel*" } | Import-Module
    }
}
```

### Step template guidelines

* Is the template a minor variation on an existing one? If so, please consider improving the existing template if possible.
* Is the name of the template consistent with the examples already in the library, in style ("Noun - Verb"), layout and casing?
* Are all parameters in the template consistent with the examples here, including help text documented with Markdown?
* Is the description of the template complete, correct Markdown?
* Is the `.json` filename consistent with the name of the template?
* Do scripts in the template validate required arguments and fail by returning a non-zero exit code when things go wrong?
* Do scripts in the template produce worthwhile status messages as they execute?
* Are you happy to contribute your template under the terms of the [license](https://github.com/OctopusDeploy/Library/blob/master/LICENSE)? If you produced the template while working for your employer please obtain written permission from them before submitting it here.
* Are the default values of parameters validly applicable in other user's environments? Don't use the default values as examples if the user will have to change them
* For how to deal with parameters and testing take a look at the article [Making great Octopus PowerShell step templates](https://www.daniellittle.xyz/making-great-octopus-powershell-step-templates/)
* For another example of how to test your step template script body before submitting a PR take a look at this [gist](https://gist.github.com/JCapriotti/45639e06ba777ee974b1)

_Before submitting your PR, please delete everything above the line below._

---

### Step template checklist

- [x] `Id` should be a **GUID** that is not `00000000-0000-0000-0000-000000000000`
  - **NOTE** If you are modifying an existing step template, please make sure that you **do not** modify the `Id` property *(updating the `Id` will break the Library sync functionality in Octopus)*. 
- [x] `Version` should be incremented, otherwise the integration with Octopus won't update the step template correctly
- [x] Parameter names should not start with `$`
- [x] **Step template parameter names (the ones declared in the JSON, not the script body) should be prefixed with a namespace so that they are less likely to clash with other user-defined variables in Octopus** (see [this issue](https://github.com/OctopusDeploy/Issues/issues/2126)). For example, use an abbreviated name of the step template or the category of the step template).
- [x] `LastModifiedBy` field must be present, and (_optionally_) updated with the correct author
- [ ] If a new `Category` has been created:
   - [ ] An image with the name `{categoryname}.png` must be present under the `step-templates/logos` folder
   - [ ] The `switch` in the `humanize` function in [`gulpfile.babel.js`](https://github.com/OctopusDeploy/Library/blob/master/gulpfile.babel.js#L92) must have a `case` statement corresponding to it

Fixes # . _If there is an open issue that this PR fixes add it here, otherwise just remove this line_

harrisonmeister commented:
Hey @phillip-haydon 

Thanks for the contribution!

I have a few comments about the change:

1. Could we change the following:

```ps
if (!(Get-Module -ListAvailable | Where-Object {  $_.Name -like "AWSPowerShel*" })) {
    	Write-Host "AWSPowerShell / AWSPowerShell.NetCore not found"
}
```

to include a non-zero exit. Essentially the previous version of just `Import-Module AWSPowerShell` would throw an error like:

```ps
Import-Module: The specified module 'AWSPowerShell' was not loaded because no valid module file was found in any module directory.
```

You can choose several ways to exit but something like this would do:

```ps
if (!(Get-Module -ListAvailable | Where-Object {  $_.Name -like "AWSPowerShel*" })) {
    	Write-Error "AWSPowerShell / AWSPowerShell.NetCore not found"
    	return
}
```

2. This is an observation (not a request to change anything). You could simplify the calls to `Get-Module -ListAvailable ...` to assign the result to a variable, and then test that, and later pipe it to `Import-Module`. 

3. Was there a reason you only used `AWSPowerShel*` and not the last l e.g. `$_.Name -like "AWSPowerShell*"

phillip-haydon commented:
1) Yup absolutely, good idea. 
2) Updated

Updated to:

```
if (Get-Module | Where-Object { $_.Name -like "AWSPowerShell*" }) {
	Write-Host "AWS PowerShell module is already loaded."
} else {
	$awsModule = Get-Module -ListAvailable | Where-Object {  $_.Name -like "AWSPowerShell*" }
    if (!($awsModule)) {
    	Write-Error "AWSPowerShell / AWSPowerShell.NetCore not found"
        return
    } else {
    	Import-Module $awsModule.Name
        Write-Host "Imported Module: $($awsModule.Name)"
    }
}
```

I ran the code twice to simulate module already loaded:

![image](https://user-images.githubusercontent.com/895629/168716489-a371a522-42ef-4fc0-9ee4-e1bfd43460b9.png)

And I removed the module to simulate a fail to load:

![image](https://user-images.githubusercontent.com/895629/168716790-86f1bc3a-3ac8-4483-a547-43cc8934696b.png)

3) Ah, didn't even notice, I prob accidently selected the `l` when I went to put `*` in and just didn't realise. Fixed.

harrisonmeister reviewed (approved):
LGTM
